File: DecompiledGameScripts/cyberpunk/devices/vendingMachines/weaponVendingMachine.swift

public class WeaponVendingMachine extends VendingMachine {

  protected let m_bigAdScreen: wref<IWorldWidgetComponent>;

  protected cb func OnRequestComponents(ri: EntityRequestComponentsInterface) -> Bool {
    super.OnRequestComponents(ri);
    EntityRequestComponentsInterface.RequestComponent(ri, n"bigAdvertScreen", n"worlduiWidgetComponent", false);
  }

  protected cb func OnTakeControl(ri: EntityResolveComponentsInterface) -> Bool {
    super.OnTakeControl(ri);
    this.m_bigAdScreen = EntityResolveComponentsInterface.GetComponent(ri, n"bigAdvertScreen") as worlduiWidgetComponent;
    this.m_controller = EntityResolveComponentsInterface.GetComponent(ri, n"controller") as WeaponVendingMachineController;
  }

  public func GetVendorID() -> TweakDBID {
    return this.GetDevicePS().GetVendorTweakID();
  }

  protected func PlayItemFall() -> Void {
    GameObject.PlaySoundEvent(this, this.GetDevicePS().GetGunFallSFX());
  }

  protected func GetJunkItem() -> ItemID {
    return ItemID.FromTDBID(this.GetDevicePS().GetJunkItemID());
  }

  protected func GetProcessingSFX() -> CName {
    return this.GetDevicePS().GetProcessingSFX();
  }

  protected func CreateDispenseRequest(shouldPay: Bool, item: ItemID) -> ref<DispenseRequest> {
    let dispenseRequest: ref<DispenseRequest> = new DispenseRequest();
    dispenseRequest.owner = this;
    dispenseRequest.position = this.RandomizePosition();
    dispenseRequest.shouldPay = shouldPay;
    if ItemID.IsValid(item) {
      dispenseRequest.itemID = ItemID.CreateQuery(ItemID.GetTDBID(item));
    };
    return dispenseRequest;
  }

  protected func TurnOffDevice() -> Void {
    super.TurnOffDevice();
    if IsDefined(this.m_bigAdScreen) {
      this.m_bigAdScreen.Toggle(false);
    };
  }

  protected func TurnOnDevice() -> Void {
    super.TurnOnDevice();
    if IsDefined(this.m_bigAdScreen) {
      this.m_bigAdScreen.Toggle(true);
    };
  }

  protected const func GetController() -> ref<WeaponVendingMachineController> {
    return this.m_controller as WeaponVendingMachineController;
  }

  public const func GetDevicePS() -> ref<WeaponVendingMachineControllerPS> {
    return this.GetController().GetPS();
  }

  public const func DeterminGameplayRole() -> EGameplayRole {
    return EGameplayRole.Distract;
  }
}

File: Definitions/Game_Definitions/classes/WeaponVendingMachine.lua
---@meta
---@diagnostic disable

---@class WeaponVendingMachine : VendingMachine
---@field bigAdScreen IWorldWidgetComponent
WeaponVendingMachine = {}

---@return WeaponVendingMachine
function WeaponVendingMachine.new() return end

---@param props table
---@return WeaponVendingMachine
function WeaponVendingMachine.new(props) return end

---@param ri entEntityRequestComponentsInterface
---@return Bool
function WeaponVendingMachine:OnRequestComponents(ri) return end

---@param ri entEntityResolveComponentsInterface
---@return Bool
function WeaponVendingMachine:OnTakeControl(ri) return end

---@param shouldPay Bool
---@param item ItemID
---@return DispenseRequest
function WeaponVendingMachine:CreateDispenseRequest(shouldPay, item) return end

---@return EGameplayRole
function WeaponVendingMachine:DeterminGameplayRole() return end

---@return WeaponVendingMachineController
function WeaponVendingMachine:GetController() return end

---@return WeaponVendingMachineControllerPS
function WeaponVendingMachine:GetDevicePS() return end

---@return ItemID
function WeaponVendingMachine:GetJunkItem() return end

---@return CName
function WeaponVendingMachine:GetProcessingSFX() return end

---@return TweakDBID
function WeaponVendingMachine:GetVendorID() return end

function WeaponVendingMachine:PlayItemFall() return end

function WeaponVendingMachine:TurnOffDevice() return end

function WeaponVendingMachine:TurnOnDevice() return end

